src/RazorSlices/RazorSlice.ResolveAndCreate.cs

```diff
@@ -87,9 +88,19 @@ private static List<SliceDefinition> LoadSlices(Assembly assembly)
 
         foreach (var rci in rcis)
         {
-            IEnumerable<PropertyInfo> properties = rci.Type.GetProperties()
+            IEnumerable<PropertyInfo> injectableProperties = rci.Type.GetProperties()
                 .Where(property => property.IsDefined(typeof(RazorInjectAttribute)) && !ExcludedServiceNames.Contains(property.PropertyType.Name));
-            var sliceDefinition = new SliceDefinition(rci.Identifier, rci.Type, GetSliceFactory(rci.Type), properties);
+            Delegate sliceFactory;
+            if (injectableProperties.Any())
```

Comment:
To avoid the ambiguity of the cost of `Any()` let's store a `bool` flag on `SliceDefinition` to make this a simple boolean check.